js/views/modals/purchase/Payment.js

```diff
@@ -1,29 +1,44 @@
+/*
```

Comment:
Made a few tweaks to this view to make it more friendly to be re-used in multiple areas, notably:
- Made the inputs more explicit. It had been relying on two models to be passed in, but the Order Detail overlay does not have the same models (at least with the same structure), so I switched it were you just directly pass in the data it needs as options.
- Updated the view so it handles partial payments. In case someone (probably accidentally) makes a partial payment from an external wallet, the totals will be updated to reflect that and if they want to pay the rest from the internal wallet (Pay From Wallet action), it will only send the remaining balance.